CHANGES.md

```diff
@@ -1,38 +1,97 @@
+# 2.17.0 - 26 Sep 2024
+
+## Features
+
+* optionally install Scenario with `ops[testing]` and expose the names in ops.testing in (#1381)
+* add a JujuVersion property for Pebble log forwarding to Loki in (#1370)
+* expand the secret ID out to the full URI when only given the ID in (#1358)
+* pre-emptively raise `InvalidStatusError` instead of waiting for Juju:
+    * make it an error to call `CollectStatusEvent.add_status` with error or unknown in (#1386)
+    * document and validate settable status values in `_ModelBackend.set_status` in (#1354)
+
+## Fixes
+
+* rework `ops.main` type hints to allow different flavours (callable class) in (#1345)
+* fix type of `StatusBase` subclasses by calling `StatusBase.register` in `__init_subclass__` in (#1383)
+* `Secret.set_info` and `Secret.set_content` can be called in the same hook in (#1373)
+
+## Documentation
+
+* add top-level intro and module-level intros in (#1320)
+* update the links to the Pebble docs in (#1362)
+* note about repeatedly setting secret value in Juju 3.6 in (#1366)
+* `config-changed` is triggered by Juju trust in (#1357)
+* typo on `CharmBase` inheritance example by @theofpa in (#1349)
+
+## Continuous Integration
+
+* periodically run the unit tests of all GitHub-hosted published charms in (#1365)
+* update the TIOBE reporting for the changes in coverage calculation in (#1367)
+* spell-check the code as part of linting in (#1388)
+* run the smoke tests on a schedule in (#1387)
+
+## Testing
+
+* properly mock the environment in (#1385)
+
+## Refactoring
+
+* move the content of `ops.testing` to `ops._private.harness` in (#1369)
+* keep the `unittest.mock` names in the 'mock' namespace in (#1379)
+* deprecate `StatusBase.register` decorator in (#1384)
+
+## Chores
+
+* note juju version on legacy workaround in (#1355)
+* re-enable test now that Pebble directory permissions are fixed in (#1363)
+* generate warnings for events that will be removed in Juju 4.0 in (#1374)
+* update charm pins in (#1390)
+* update charm pins in (#1375)
+* bump version in (#1348)
+* bump canonical/setup-lxd version in (#1350)
+* bump actions/attest-build-provenance from 1.4.0 to 1.4.2 in (#1351)
```

Comment:
My preference is that housecleaning stuff like this is not included (you can read the git log for that). I think that this is the team decision also, although someone can correct me if not.

Reply:
I noticed that other releases didn't include these items, but I also didn't see anything about removing them in HACKING.md. I'm happy to remove them.

Reply:
Which I've now done

Reply:
> I noticed that other releases didn't include these items, but I also didn't see anything about removing them in HACKING.md.

Good point. Probably should be part of:

>> 6. Where appropriate, collapse multiple tightly related bullet points into a single point that refers to multiple commits.

Feel free to open a PR for HACKING.md improvements as well - it's always most insightful when people do their first release.